bisection.py

```diff
@@ -0,0 +1,373 @@
+TORCHBENCH_GITREPO="https://github.com/pytorch/benchmark.git"
+TORCHBENCH_SCORE_CONFIG="torchbenchmark/score/configs/v0/config-v0.yaml"
+TORCHBENCH_DEPS = {
+    "torchtext": os.path.expandvars("${HOME}/text"),
+    "torchvision": os.path.expandvars("${HOME}/vision"),
+    "torchaudio": os.path.expandvars("${HOME}/audio"),
+}
+
+class Commit:
+    sha: str
+    ctime: str
+    score: Optional[float]
+    def __init__(self, sha, ctime, score):
+        self.sha = sha
+        self.ctime = ctime
+        self.score = score
+    def __str__(self):
+        return self.sha
+
+class TorchSource:
+    srcpath: str
+    commit_date: datetime
+    commits: List[Commit]
+    # Map from commit SHA to index in commits
+    commit_dict: Dict[str, int]
+    def __init__(self, srcpath: str):
+        self.srcpath = srcpath
+        self.commits = []
+        self.commit_dict = dict()
+
+    def prep(self) -> bool:
+        repo_origin_url = gitutils.get_git_origin(self.srcpath)
+        if not repo_origin_url == TORCH_GITREPO:
+            print(f"Unmatched repo origin url: {repo_origin_url} with standard {TORCH_GITREPO}")
+            return False
+        return True
+    
+    # Get all commits between start and end, save them in commits
+    def init_commits(self, start: str, end: str) -> bool:
+        commits = gitutils.get_git_commits(self.srcpath, start, end)
+        if not commits or len(commits) < 2:
+            print(f"Failed to retrieve commits from {start} to {end} in {self.srcpath}.")
+            return False
+        for count, commit in enumerate(commits):
+            ctime = gitutils.get_git_commit_date(self.srcpath, commit)
+            self.commits.append(Commit(sha=commit, ctime=ctime, score=None))
+            self.commit_dict[commit] = count
+        # Setup commit date
+        last_commit_date = self.commits[-1].ctime.split()[0]
+        self.commit_date = datetime.strptime(last_commit_date, "%Y-%m-%d")
+        return True
+    
+    def get_mid_commit(self, left: Commit, right: Commit) -> Optional[Commit]:
+        left_index = self.commit_dict[left.sha]
+        right_index = self.commit_dict[right.sha]
+        if right_index == left_index + 1:
+            return None
+        else:
+            return self.commits[(left_index + right_index) / 2]
+
+    def setup_build_env(self, env):
+        env["USE_CUDA"] = "1"
+        env["BUILD_CAFFE2_OPS"] = "0"
+        env["USE_XNNPACK"] = "0"
+        env["USE_MKLDNN"] = "1"
+        env["USE_MKL"] = "1"
+        env["USE_CUDNN"] = "1"
+        env["CMAKE_PREFIX_PATH"] = env["CONDA_PREFIX"]
+        return env
+
+    # Install dependencies such as torchaudio, torchtext and torchvision
+    def install_deps(self, build_env):
+        # Build torchvision
+        print(f"Building torchvision ...", end="", flush=True)
+        command = "python setup.py install &> /dev/null"
+        subprocess.check_call(command, cwd=TORCHBENCH_DEPS["torchvision"], env=build_env, shell=True)
```

Comment:
a little nit pick, i think it's better to give lists to subprocess for cmd.  e.g.
['python', 'setup.py', ..] 
as well as there are ways to let subprocess capture the output of the process for you using PIPE and not just discard to /dev/null.  I think that can be changed later, but happy to help with pointers to the right methods if you want to do it.